### PR TITLE
Order hist final

### PR DESCRIPTION
**Pull request by kevinjcai**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # "This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation."
🛍 CC: @EthanAuyeung

### CODE REVIEW

**Comment by BuyankhuuTsCAl** [comment]
make sure you are following the correct branch naming format! such as Kevin/blah blah

File: src/app/orderHistory/page.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      const ids = await fetchOrderIdsByUserIdSorted();
+      setOrderIds(ids);
+    };
+
+    fetchIds();
+  }, []);
+
+  return (
+    <div>
+      <NavBarMovedUP />
```
Comment (BuyankhuuTsCAl) [inline src/app/orderHistory/page.tsx:30]: why are you using NavBarMovedUP?

File: src/app/orderHistory/page.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      <OutterBox>
+        <BackButton destination="./profileScreen" />
+        <h1>Order History</h1>
```
Comment (BuyankhuuTsCAl) [inline src/app/orderHistory/page.tsx]: make sure to use the text components within the global folder!

File: src/app/orderHistory/styles.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+// styles.ts (for styled-components)
+import styled from 'styled-components';
+import NavBar from '../../components/NavBarFolder/NavBar';
+
+import Footer from '../../components/FooterFolder/Footer';
+
+export const FooterMoved = styled(Footer)`
```
Comment (BuyankhuuTsCAl) [inline src/app/orderHistory/styles.ts:7]: interesting? What the purpose of this again?

File: src/app/orderHistory/styles.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  width: 900px;
+  margin: 0 auto; // This will center the OutterBox
+`;
+
+export const OrderHistoryContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  border-radius: 10px;
+  background: var(--White, #fff);
```
Comment (BuyankhuuTsCAl) [inline src/app/orderHistory/styles.ts:25]: make sure that you are using the colors components from the global color file

File: src/app/orderHistory/styles.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  margin-top: 10px;
+  padding: 0; // Ensure there's no padding pushing the internal boxes inward
+`;
+
+export const OrderHistoryBox = styled.div`
+  width: 100%; // Make internal box take full width of the container
+  margin: 20px 0; // Keep vertical spacing between orders
+  padding: 30px; // Inner spacing for content inside the box
+  border-radius: 8px;
+  background: #fff;
```
Comment (BuyankhuuTsCAl) [inline src/app/orderHistory/styles.ts:39]: same here

File: src/app/orderHistory/styles.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  margin: 0 auto; // This will center the OutterBox
+`;
+
+export const OrderHistoryContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  border-radius: 10px;
+  background: var(--White, #fff);
+  box-shadow: 0px 1px 4px 1px rgba(0, 0, 0, 0.2);
```
Comment (BuyankhuuTsCAl) [inline src/app/orderHistory/styles.ts:26]: same here

File: src/app/orderHistory/styles.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  padding: 0; // Ensure there's no padding pushing the internal boxes inward
+`;
+
+export const OrderHistoryBox = styled.div`
+  width: 100%; // Make internal box take full width of the container
+  margin: 20px 0; // Keep vertical spacing between orders
+  padding: 30px; // Inner spacing for content inside the box
+  border-radius: 8px;
+  background: #fff;
+  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
```
Comment (BuyankhuuTsCAl) [inline src/app/orderHistory/styles.ts:40]: same here

File: src/app/orderHistory/styles.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+import styled from 'styled-components';
+import NavBar from '../../components/NavBarFolder/NavBar';
+
+import Footer from '../../components/FooterFolder/Footer';
+
+export const FooterMoved = styled(Footer)`
+  left: 0;
+`;
+
+export const NavBarMovedUP = styled(NavBar)`
```
Comment (BuyankhuuTsCAl) [inline src/app/orderHistory/styles.ts:11]: can you double check that the position of the navbar isn't already relative

File: src/app/orderHistory/styles.ts [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  position: relative;
+`;
+
+export const OutterBox = styled.div`
+  width: 900px;
+  margin: 0 auto; // This will center the OutterBox
```
Comment (BuyankhuuTsCAl) [inline src/app/orderHistory/styles.ts:17]: if you are gonna center, it might be cleaner to use flexbox?

File: src/app/orderHistory/styles.ts (hunk 5 above)
Comment (BuyankhuuTsCAl) [reply src/app/orderHistory/styles.ts:39]: a little confused on why both orderhistorybox and orderhistorycontainer are both flex boxes that are scrollable. What is the difference and do you mind changing the name to something that would help me understand the difference since in some cases a box is a container

File: src/app/pickup/page.tsx [hunk 10]
```diff
@@ -31,25 +33,67 @@ import {
   PickupTimeButton,
 } from './styles';
 
+function DateInfoComponent(date: { date: unknown }) {
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx:35]: I think you Monique and Charlotte have similar function! if you don't mind can you put this in a global folder and call it

File: src/app/pickup/page.tsx [hunk 11]
```diff
@@ -31,25 +33,67 @@ import {
...
+      const data = await fetchNRecentPickupTimes(2); // change the function to grab the cartItems as products
       setTimes(data);
-      console.log(Time);
     }
     fetchProducts();
     fetchTimes();
   }, []);
 
+  useEffect(() => {
+    async function fetchUserData() {
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx:82]: why is there two useeffects without a dependency?

File: src/app/pickup/page.tsx [hunk 12]
```diff
@@ -65,12 +109,39 @@ export default function Pickup() {
...
             <Normal700Text>Name</Normal700Text>
-            <PickupContent>Ethan Auyeung</PickupContent>
+            <PickupContent>
+              {Profile?.first_name} {Profile?.last_name}
+            </PickupContent>
             <Normal700Text>Phone Number</Normal700Text>
-            <PickupContent>+1 123-456-7890</PickupContent>
-            <PickupButton day="hi" date="date" start="start" end="end" />
+            <PickupContent>{Profile?.phone_numbers}</PickupContent>
+            <div style={{ display: 'flex', justifyContent: 'space-around' }}>
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx]: don't use inline styling! create a new styled component and calll it

File: src/app/pickup/page.tsx [hunk 13]
```diff
@@ -65,12 +109,39 @@ export default function Pickup() {
...
+              <Normal700Text>Pick Up Time</Normal700Text>
+              <Normal700Text>Pick Up times: 10:00 AM - 12:00 PM </Normal700Text>
+            </div>
+
+            <div style={{ display: 'flex', justifyContent: 'space-around' }}>
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx]: same thing

File: src/app/pickup/page.tsx [hunk 14]
```diff
@@ -65,12 +109,39 @@ export default function Pickup() {
...
+              {Time.map((time, index) => (
+                <PickupTimeButton
+                  key={time.id} // Replace 'index' with a unique identifier from the 'time' object
+                  isSelected={selectedPickupIndex === index}
+                  onClick={() => handleButtonClick(index)}
+                >
+                  <div>
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx:123]: if this is the date, why is it surrounded by a div and why is there no styling? try to use the text components

File: src/app/pickup/page.tsx [hunk 15]
```diff
@@ -65,12 +109,39 @@ export default function Pickup() {
...
+                    {String(
+                      DateInfoComponent({ date: time.start_time })
+                        ?.getDayOfWeek,
+                    )}
+                  </div>
+                  <div>
+                    {
+                      DateInfoComponent({ date: time.start_time })
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx:131]: same idea

File: src/app/pickup/page.tsx [hunk 16]
```diff
@@ -91,15 +162,34 @@ export default function Pickup() {
               <HeaderShiftRight
               // change with the actual cart total
               >
-                10
+                {Cart.reduce((acc, item) => acc + item.quantity, 0)}
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx:159]: wait what is this?

Comment (BuyankhuuTsCAl) [reply src/app/pickup/page.tsx:159]: if this is cart total, isn't there a query that i wrote for it?

File: src/app/pickup/page.tsx [hunk 17]
```diff
@@ -91,15 +162,34 @@ export default function Pickup() {
...
+            onClick={async () => {
+              // Add the pickup ID to the order
+              const pickupId =
+                selectedPickupIndex !== null
+                  ? Time[selectedPickupIndex]?.id
+                  : null;
+              if (pickupId) {
+                await updateCartPickupId(pickupId);
+
+                // Add your code here to update the order with the pickup ID
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx]: can u get rid of these comments?

File: src/app/pickup/styles.ts [hunk 18]
```diff
@@ -30,7 +30,7 @@ export const PickupContent = styled.div`
 `;
 
 export const PickupTimeButton = styled.button`
-  background: ${COLORS.lightGrey};
+  background: {COLORS.periwinkle};
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/styles.ts]: I got rid of the previous code that you wrote because the syntax was wrong. What you want to do is pass in a parameter to this! There is a ton of previous examples in different files and determine which colors you can use from that

File: src/components/OrderHistory/OrderHistoryBox.tsx [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchData();
+  }, [orderId]);
+
+  if (loading) {
+    return <div>Loading...</div>;
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/OrderHistoryBox.tsx:42]: why is text surrounded by div?

File: src/components/OrderHistory/OrderHistoryBox.tsx [hunk 20]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  if (!order) {
+    return <div>Order not found.</div>;
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/OrderHistoryBox.tsx]: same here

File: src/components/OrderHistory/OrderHistoryText.tsx [hunk 21]
```diff
@@ -0,0 +1,54 @@
...
+
+  const viewOrder = (orderID: string) => {
+    const queryString = querystring.stringify({ orderID });
+    router.push(`/orderPage?${queryString}`);
+  };
+
+  return (
+    <div
+      style={{
+        display: 'flex',
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/OrderHistoryText.tsx]: don't use in line styling!

File: src/components/OrderHistory/OrderHistoryText.tsx [hunk 22]
```diff
@@ -0,0 +1,54 @@
...
+        justifyContent: 'space-between',
+        alignItems: 'center',
+        marginBottom: '20px',
+      }}
+    >
+      <div>
+        <div>{formatDate(date)}</div>
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/OrderHistoryText.tsx]: too many divs! if its text, it should be surrounded by a styled text componet from the global folder

File: src/components/OrderHistory/OrderHistoryText.tsx [hunk 23]
```diff
@@ -0,0 +1,54 @@
...
+        <div>Order No. {orderNumber}</div>
+        <div style={{ color: status === 'Confirmed' ? 'green' : 'red' }}>
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/OrderHistoryText.tsx]: this wasn't working btw when I looked at the page!

File: src/components/OrderHistory/OrderHistoryText.tsx [hunk 24]
```diff
@@ -0,0 +1,54 @@
...
+          {status === 'Confirmed' ? '✓' : '✗'} {status}
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/OrderHistoryText.tsx]: use icons not text

File: src/components/OrderHistory/OrderHistoryText.tsx [hunk 25]
```diff
@@ -0,0 +1,54 @@
...
+        </div>
+      </div>
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/OrderHistoryText.tsx:163]: why is it nested divs?

File: src/components/OrderHistory/OrderHistoryText.tsx [hunk 26]
```diff
@@ -0,0 +1,54 @@
...
+      <ViewOrderButton type="button" onClick={() => viewOrder(orderNumber)}>
+        {/** DO NOT USE IMAGE Please use the icon in the feather library! */}
+        View order <ArrowIcon />
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/OrderHistoryText.tsx]: text component!

File: src/components/OrderHistory/styles.ts [hunk 27]
```diff
@@ -0,0 +1,98 @@
...
+import styled from 'styled-components';
+import COLORS from '../../styles/colors';
+
+export const containerStyle = styled.div`
+  width: 250px; // This width might be too small if it's meant to contain the OrderDiv
+  height: 250px;
+  margin-right: auto; // Center the container if that's the intention
+  margin-left: auto; // Center the container if that's the intention
+  color: ${COLORS.black};
+  font-family: Public Sans;
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/styles.ts:12]: this can be removed once you add a text components

File: src/components/OrderHistory/styles.ts [hunk 28]
```diff
@@ -0,0 +1,98 @@
+// styles.ts
+import styled from 'styled-components';
+import COLORS from '../../styles/colors';
+
+export const containerStyle = styled.div`
+  width: 250px; // This width might be too small if it's meant to contain the OrderDiv
+  height: 250px;
+  margin-right: auto; // Center the container if that's the intention
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/styles.ts:9]: if you are trying to center please use flex box

File: src/components/OrderHistory/styles.ts [hunk 29]
```diff
@@ -0,0 +1,98 @@
...
+export const imageGalleryStyle = {
+  display: 'flex',
+  overflowX: 'auto',
+  gap: '10px',
+  // Add more styles as needed
+};
+
+export const viewButtonStyle = {
+  marginTop: '10px',
+  backgroundColor: 'blue',
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/styles.ts:27]: colors should be grabbed from colors file

File: src/components/OrderHistory/styles.ts [hunk 30]
```diff
@@ -0,0 +1,98 @@
...
+
+export const imageGalleryStyle = {
+  display: 'flex',
+  overflowX: 'auto',
+  gap: '10px',
+  // Add more styles as needed
+};
+
+export const viewButtonStyle = {
+  marginTop: '10px',
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/styles.ts:26]: it should be margin-top

File: src/components/OrderHistory/styles.ts (hunk 29 above)
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/styles.ts:27]: same idea background-color i believe

File: src/components/OrderHistory/styles.ts [hunk 31]
```diff
@@ -0,0 +1,98 @@
...
+  backgroundColor: 'blue',
+  color: 'white',
+  padding: '10px 20px',
+  cursor: 'pointer',
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/styles.ts:30]: what's this for?

File: src/components/OrderHistory/styles.ts [hunk 32]
```diff
@@ -0,0 +1,98 @@
...
+  // Add more styles as needed
+};
+
+export const ViewOrderButton = styled.button`
+  overflow: hidden;
+  color: var(--Black, #101010);
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/styles.ts:36]: grab teh color from file

File: src/components/OrderHistory/styles.ts [hunk 33]
```diff
@@ -0,0 +1,98 @@
...
+  text-align: right;
+  text-overflow: ellipsis;
+  font-family: 'Public Sans', sans-serif;
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/styles.ts]: text will taken from teh file so you should be able to delete it here and just use the text compoentn

File: src/components/OrderHistory/styles.ts [hunk 34]
```diff
@@ -0,0 +1,98 @@
...
+  text-decoration-line: underline;
+  background: none;
+  border: none;
+  cursor: pointer;
+  display: flex;
+  align-items: center;
+`;
+
+export const ArrowIcon = styled.div`
+  width: 0;
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/styles.ts:51]: wait huh? doesn't mean that this doesn't exist if the height and width are 0?

**Review by BuyankhuuTsCAl — COMMENTED** [review]
Hey Kevin, it is looking close to being done! there is still alot of previous comments that haven't been resolved and some issues with the pages. please look at them and get them fixed asap!

File: src/components/OrderHistory/OrderHistoryText.tsx (hunk 24 above)
Comment (BuyankhuuTsCAl) [reply src/components/OrderHistory/OrderHistoryText.tsx]: also there's three status for orders! such as confirmed, rejected and in progress. Make sure to look at figma for the correct icon to use

File: src/app/pickup/page.tsx [hunk 35]
```diff
@@ -31,46 +32,111 @@ import {
...
+  const [Cart, setCart] = useState<ProductWithQuantity[]>([]);
   const router = useRouter();
   const [Time, setTimes] = useState<Pickup[]>([]);
-
+  const [Profile, setProfile] = useState<User>();
+  const [selectedPickupIndex, setSelectedPickupIndex] = useState<number>(0);
   useEffect(() => {
     async function fetchProducts() {
-      const data = await arrayOfFavorites(); // change the function to grab the cartItems as products
+      const data = await fetchCartItemsWithQuantity(); // change the function to grab the cartItems as products
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx:70]: maybe use a different word then data

File: src/app/pickup/page.tsx [hunk 36]
```diff
@@ -31,46 +32,111 @@ import {
...
       setCart(data);
     }
     async function fetchTimes() {
-      const data = await fetchRecentPickupTimes(); // change the function to grab the cartItems as products
+      const data = await fetchNRecentPickupTimes(2); // change the function to grab the cartItems as products
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx:74]: same here

File: src/app/pickup/page.tsx [hunk 37]
```diff
@@ -31,46 +32,111 @@ import {
...
       setTimes(data);
-      console.log(Time);
     }
     fetchProducts();
     fetchTimes();
   }, []);
 
+  useEffect(() => {
+    async function fetchUserData() {
+      const data = await fetchUser(); // change the function to grab the cartItems as products
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx:83]: same here

File: src/app/pickup/page.tsx [hunk 38]
```diff
@@ -31,46 +32,111 @@ import {
...
+    setSelectedPickupIndex(index);
+  };
+
   return (
     <div>
-      <NavBarMovedUP />
-
+      <NavBar />
       <PageDiv>
         <ForceColumnDiv>
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx:97]: can you rename this to left column

File: src/app/pickup/page.tsx [hunk 39]
```diff
@@ -31,46 +32,111 @@ import {
...
           <BackDiv onClick={() => router.push('/cart')}>
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx:98]: use the backButton component

File: src/app/pickup/page.tsx [hunk 40]
```diff
@@ -31,46 +32,111 @@ import {
...
-            </Normal700Text>
-            <Normal700Text>Name</Normal700Text>
-            <PickupContent>Ethan Auyeung</PickupContent>
-            <Normal700Text>Phone Number</Normal700Text>
-            <PickupContent>+1 123-456-7890</PickupContent>
-            <PickupButton day="hi" date="date" start="start" end="end" />
+            </Heading4Bold>
+            <Heading4Bold>Name</Heading4Bold>
+            <PickupContent>
+              {Profile?.first_name} {Profile?.last_name}
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx:108]: make sure to use textComponents similar to the example above!

File: src/app/pickup/page.tsx [hunk 41]
```diff
@@ -31,46 +32,111 @@ import {
...
+            </PickupContent>
+            <Heading4Bold>Phone Number</Heading4Bold>
+            <PickupContent>{Profile?.phone_numbers}</PickupContent>
+            <div style={{ display: 'flex', justifyContent: 'space-between' }}>
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx:112]: can you create a styled div component in style and render that here?

File: src/app/pickup/page.tsx [hunk 42]
```diff
@@ -31,46 +32,111 @@ import {
...
+              <Heading4Bold>Time Slot</Heading4Bold>
+              <Normal700Text>Pick Up times: 10:00 AM - 12:00 PM </Normal700Text>
+            </div>
+
+            <div style={{ display: 'flex', justifyContent: 'space-between' }}>
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx:116]: can you create a styled div component in style and render that here?

File: src/app/pickup/page.tsx [hunk 43]
```diff
@@ -31,46 +32,111 @@ import {
...
+              {Time.map(time => (
+                <PickupTimeButton
+                  key={time.id}
+                  $isSelected={selectedPickupIndex === time.id}
+                  onClick={() => handleButtonClick(time.id)}
+                >
+                  <div>
+                    {String(
+                      DateInfoComponent({ date: time.start_time })
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx:125]: make sure that this text is getting styled!

File: src/app/pickup/page.tsx [hunk 44]
```diff
@@ -31,46 +32,111 @@ import {
...
+                        ?.dateAsMonthDay
+                    }
+                  </div>
+                </PickupTimeButton>
+              ))}
+            </div>
 
-            <PickupTimeButton>Time</PickupTimeButton>
             <div>Location: 3170 23rd Street, San Francisco, CA 94110</div>
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx:139]: style the location and don't surround it with a regular div

File: src/app/pickup/page.tsx [hunk 45]
```diff
@@ -91,15 +157,22 @@ export default function Pickup() {
...
           </WhiteBackgroundDiv>
 
           <CheckoutButton
-            onClick={() => router.push('/orderConfirmationPickUp')}
+            onClick={async () => {
+              if (selectedPickupIndex !== 0) {
+                await updateCartPickupId(selectedPickupIndex);
+                router.push('/orderConfirmationPickUp');
+              } else {
+                // handle the case where they didn't select a time!
```
Comment (BuyankhuuTsCAl) [inline src/app/pickup/page.tsx]: please implement this functionality

File: src/components/OrderHistory/ImageCarousel.tsx [hunk 46]
```diff
@@ -0,0 +1,21 @@
...
+  images: string[]; // Array of image URLs
+}
+
+export default function ImageGallery(props: ImageGalleryProps) {
+  const { images } = props;
+
+  return (
+    <GalleryContainer>
+      {images.map((image, index) => (
+        <ImageContainer key={index}>
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/ImageCarousel.tsx:15]: usually you don't use index as key because if they change the index the keys get messed up

File: src/components/OrderHistory/OrderHistoryText.tsx [hunk 47]
```diff
@@ -0,0 +1,167 @@
...
+  const router = useRouter();
+
+  const viewOrder = (orderID: string) => {
+    const queryString = querystring.stringify({ orderID });
+    router.push(`/orderPage?${queryString}`);
+  };
+  if (order.status === OrderStatus.OrderRejected) {
+    return (
+      <div
+        style={{
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/OrderHistoryText.tsx:46]: create a separate styled component and render it here

File: src/components/OrderHistory/OrderHistoryText.tsx [hunk 48]
```diff
@@ -0,0 +1,167 @@
...
+          justifyContent: 'space-between',
+          alignItems: 'start',
+          marginBottom: '20px',
+          flexDirection: 'column',
+        }}
+      >
+        <div
+          style={{
+            marginTop: '10px',
+            display: 'flex',
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/OrderHistoryText.tsx:57]: same idea here

File: src/components/OrderHistory/OrderHistoryText.tsx [hunk 49]
```diff
@@ -0,0 +1,167 @@
...
+        </OrderStatusDiv>
+      </div>
+    );
+  }
+  if (order.status === OrderStatus.OrderApproved) {
+    return (
+      <div
+        style={{
+          display: 'flex',
+          justifyContent: 'space-between',
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/OrderHistoryText.tsx:89]: same here

File: src/components/OrderHistory/OrderHistoryText.tsx [hunk 50]
```diff
@@ -0,0 +1,167 @@
...
+          alignItems: 'start',
+          marginBottom: '20px',
+          flexDirection: 'column',
+        }}
+      >
+        <div
+          style={{
+            marginTop: '10px',
+            display: 'flex',
+            flexDirection: 'row',
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/OrderHistoryText.tsx:99]: same here

File: src/components/OrderHistory/styles.ts [hunk 51]
```diff
@@ -0,0 +1,145 @@
...
+    text-decoration: underline;
+  }
+`;
+
+export const ArrowIcon = styled.div`
+  width: 0;
+  height: 0;
+  border-top: 5px solid transparent;
+  border-bottom: 5px solid transparent;
+  border-left: 5px solid var(--Black, #101010);
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/styles.ts:55]: grab color from COLORS

File: src/components/OrderHistory/styles.ts [hunk 52]
```diff
@@ -0,0 +1,145 @@
...
+export const GalleryContainer = styled.div`
+  display: flex;
+  overflow-x: auto;
+  gap: 10px;
+  padding: 10px;
+`;
+
+export const ImageContainer = styled.div`
+  flex: 0 0 auto;
+  background-color: var(--Greyish, #e6e6e6);
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/styles.ts:152]: grab color

File: src/components/OrderHistory/styles.ts [hunk 53]
```diff
@@ -0,0 +1,145 @@
...
+export const RowDiv = styled.div`
+  width: 700px;
+  height: 400px;
+  margin-bottom: 50px;
+`;
+
+export const OrderStatusDiv = styled.div`
+  width: 300px;
+  height: 35px;
+  background: var(--Light-Red, #fdd);
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/styles.ts:175]: create/grab color

File: src/components/OrderHistory/styles.ts [hunk 54]
```diff
@@ -0,0 +1,145 @@
...
+  justify-content: center;
+  align-items: center;
+  border-radius: 20px;
+  margin-top: 15px;
+`;
+
+export const OrderStatusApprovedDiv = styled.div`
+  width: 300px;
+  height: 35px;
+  background: var(--Lime-Green, #cee8be);
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/styles.ts:186]: same here

File: src/components/OrderHistory/styles.ts [hunk 55]
```diff
@@ -0,0 +1,145 @@
...
+  justify-content: center;
+  align-items: center;
+  border-radius: 20px;
+  margin-top: 15px;
+`;
+
+export const OrderStatusSubmittedDiv = styled.div`
+  width: 300px;
+  height: 35px;
+  background: var(--Baby-Blue, #c7ddff);
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderHistory/styles.ts:197]: same here

File: src/components/OrderSummaryFolder/OrderSummary.tsx [hunk 56]
```diff
@@ -22,18 +21,26 @@ export default function OrderSummary(props: {
...
+
+              <Body1>{cartItem.quantity}</Body1>
+            </ItemSummaryDiv>
+          ))}
+        </OrderSummaryDiv>
+      </AlignItemCenter>
       <OrderTotalDiv>
-        <HeaderShiftLeft>Order Total</HeaderShiftLeft>
+        <Body1Bold style={{ marginLeft: '10px' }}>Order Total</Body1Bold>
         <HeaderShiftRight>{numberOfItems}</HeaderShiftRight>
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderSummaryFolder/OrderSummary.tsx:44]: style this text

File: src/components/OrderSummaryFolder/styles.ts [hunk 57]
```diff
@@ -2,39 +2,39 @@ import styled from 'styled-components';
...
 `;
 
 export const OrderTotalDiv = styled.div`
   height: 50px;
-  padding-top: 10px;
-  width: 350px;
-  padding-left: 0px;
+  padding-top: 5px;
+  width: 100%;
   border-top: 1px black solid;
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderSummaryFolder/styles.ts:23]: color

File: src/components/OrderSummaryFolder/styles.ts [hunk 58]
```diff
@@ -2,39 +2,39 @@ import styled from 'styled-components';
...
 `;
 
 export const Qty = styled.p`
   width: 100%;
-  padding-left: 290px;
 `;
 
 export const WhiteBackgroundDiv = styled.div`
   border-radius: 8px;
   background: var(--White, #fff);
```
Comment (BuyankhuuTsCAl) [inline src/components/OrderSummaryFolder/styles.ts:36]: color

**Review by BuyankhuuTsCAl — CHANGES_REQUESTED** [review]
